Some sanitizers needs backtrace on x86_64

This has been ported from https://github.com/void-linux/void-packages/blob/c907a54de30ad0b19fbf9f37d5b67cabe5c7744d/srcpkgs/llvm12/template#L126

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -424,6 +424,10 @@
 append_list_if(COMPILER_RT_HAS_NODEFAULTLIBS_FLAG -nodefaultlibs SANITIZER_COMMON_LINK_FLAGS)
 append_list_if(COMPILER_RT_HAS_Z_TEXT -Wl,-z,text SANITIZER_COMMON_LINK_FLAGS)
 
+if("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "x86_64")
+  list(APPEND SANITIZER_COMMON_LINK_LIBS execinfo)
+endif()
+
 if (COMPILER_RT_USE_BUILTINS_LIBRARY)
   string(REPLACE "-Wl,-z,defs" "" CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}")
 else()
